extensions/reactive-routes/runtime/src/main/java/io/quarkus/vertx/web/runtime/HttpCompressionHandler.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.vertx.web.runtime;
+
+import java.util.Set;
+
+import io.quarkus.vertx.http.runtime.HttpCompression;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+
+public class HttpCompressionHandler implements Handler<RoutingContext> {
+
+    private final Handler<RoutingContext> routeHandler;
+    private final HttpCompression compression;
+    private final Set<String> compressedMediaTypes;
+
+    public HttpCompressionHandler(Handler<RoutingContext> routeHandler, HttpCompression compression,
+            Set<String> compressedMediaTypes) {
+        this.routeHandler = routeHandler;
+        this.compression = compression;
+        this.compressedMediaTypes = compressedMediaTypes;
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+        context.addEndHandler(new Handler<AsyncResult<Void>>() {
```

Comment:
To avoid that interception, we could imagine moving that code directly to the HTTP Server "root" request handler. Just an idea if we see a large decrease in terms of performance.

Reply:
Yeah, so this handler should only be used if compression is enabled and I think that there will be other much more expensive calls in such case ;-)